packages/zoe/test/unitTests/test-blockedOffers.js

```diff
@@ -0,0 +1,104 @@
+// @ts-nocheck
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { test } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import path from 'path';
+
+import { E } from '@endo/eventual-send';
+import bundleSource from '@endo/bundle-source';
+
+import { setup } from './setupBasicMints.js';
+import { makeZoeKit } from '../../src/zoeService/zoe.js';
+import { makeFakeVatAdmin } from '../../tools/fakeVatAdmin.js';
+import {
+  depositToSeat,
+  withdrawFromSeat,
+} from '../../src/contractSupport/index.js';
+import { assertPayoutAmount } from '../zoeTestHelpers.js';
+import { makeOffer } from './makeOffer.js';
+
+const filename = new URL(import.meta.url).pathname;
+const dirname = path.dirname(filename);
+
+const contractRoot = `${dirname}/zcf/zcfTesterContract.js`;
+
+const setupContract = async (moolaIssuer, bucksIssuer) => {
+  let testJig;
+  const setJig = jig => {
+    testJig = jig;
+  };
+  const fakeVatAdmin = makeFakeVatAdmin(setJig);
+  const { zoeService: zoe } = makeZoeKit(fakeVatAdmin.admin);
+
+  // pack the contract
+  const bundle = await bundleSource(contractRoot);
+  fakeVatAdmin.vatAdminState.installBundle('b1-contract', bundle);
+  // install the contract
+  const installation = await E(zoe).installBundleID('b1-contract');
+
+  // Alice creates an instance
+  const issuerKeywordRecord = harden({
+    Pixels: moolaIssuer,
+    Money: bucksIssuer,
+  });
+
+  await E(zoe).startInstance(installation, issuerKeywordRecord);
+
+  /** @type {ContractFacet} */
+  const zcf = testJig.zcf;
+  return { zoe, zcf };
+};
+
+test(`blockedOffers - deny`, async t => {
+  const { moola, moolaIssuer, bucksMint, bucks, bucksIssuer } = setup();
+  const { zoe, zcf } = await setupContract(moolaIssuer, bucksIssuer);
+  await zcf.setOfferFilter(['eat']);
+
+  await t.throwsAsync(
+    () =>
+      makeOffer(
+        zoe,
+        zcf,
+        harden({ want: { A: moola(3n) }, give: { B: bucks(5n) } }),
+        harden({ B: bucksMint.mintPayment(bucks(5n)) }),
+      ),
+    { message: 'not accepting offer with description "seat"' },
```

Comment:
That setting the offer filter to `['eat']` blocks invitations with description `'seat'` is disturbing. It is clearly accident prone. For the scenarios off the top of my head, exact match is fine. Why do you want substring? Do you have a use case in mind?

(I seem to have a vague memory that we may have discussed this and I agreed with substring. But I don't quite remember and I am confused now.)